### ISSUE TITLE
[4.x]: Conditional fields in globals aren't saved

### ISSUE THREAD

**Issue by gglnx** [craftcms/cms#12166]
### Description

After saving a global set with conditional fields the conditional fields aren't saved and stay empty. `Element::setFieldValuesFromRequest()` used in `globals/save-content` only saves to visible fields and making fields visible beforehand is missing.

### Steps to reproduce

1. Create a global set with two fields (lightswitch and text field for example)
2. Make the text field depend on the lightswitch field
3. Try to save content to the text field

### Expected behavior

Working and savable conditional fields in global sets.

### Actual behavior

Empty conditional fields after saving.

### Craft CMS version

4.2.8

### Installed plugins and versions

-

**Comment by i-just** [comment]
Hi, thanks for getting in touch. I can't seem to replicate this issue in Craft CMS 4.2.8. Could you please confirm if this is happening for you with a standalone lightswitch field and plain text field (where the plain text field is e.g. set to only show if the lightswitch is turned on) or is it happening for a more complex scenario, and if so, could you please provide a bit more info on how to replicate?

**Comment by gglnx** [comment]
I can't reproduce it either now. That's my field layout: 

![SCR-20221022-m3v](https://user-images.githubusercontent.com/23519/197343013-2c3cee67-bb56-4f9b-9292-9f8caddd79b2.png)

I removed the conditions, changed the content and added the conditions backs and now it works. Even from clean install now everything works. You can close the ticket I think.

**Comment by brandonkelly** [comment]
Weird, let us know if it crops up again.

**Comment by andygrn** [comment]
I believe I'm seeing this issue now too, but on users rather than globals. Craft v4.3.1.

My users have two required custom fields; a dropdown field, and a plain text field conditional on the dropdown value. I don't seem to ever be able to save a user when the plain text field is active, because the plain text field always comes back empty with a 'cannot be blank' error.

This appears to be because `Element::setFieldValuesFromRequest()` is running `Element::fieldLayoutFields($visibleOnly = true)` to determine which fields to set, but obviously this is happening before the dropdown field value is populated, so the conditional field is never active and its input is always lost. A bit of a chicken and egg situation, you need to set the field values to determine which field values to set.

Somehow this same setup works fine for entries though - perhaps a side effect of drafts not running full validation? I can zip up a minimal example install if you need it.

**Comment by brandonkelly** [comment]
@andygrn Yeah if you can give us specific steps to reproduce that would be helpful.

**Comment by andygrn** [comment]
@brandonkelly Thanks - I've sent a zip to support@craftcms.com.

**Comment by brandonkelly** [comment]
We got this fixed for the next release via #12326 🎉

**Comment by brandonkelly** [comment]
Craft 4.3.2 is out with that fix.
